src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -129,29 +130,61 @@ public void OnResponseHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
                         throw new Http2ProtocolException(Http2ProtocolErrorCode.ProtocolError);
                     }
 
-                    if (name.SequenceEqual(s_statusHeaderName))
+                    if (name[0] == 58) // ':'
                     {
                         if (_state == StreamState.ExpectingTrailingHeaders)
                         {
                             // Pseudo-headers not allowed in trailers.
                             if (NetEventSource.IsEnabled) _connection.Trace("Pseudo-header in trailer headers.");
                             throw new HttpRequestException(SR.net_http_invalid_response);
                         }
-
-                        if (value.Length != 3)
-                            throw new Exception("Invalid status code");
-
-                        // Copied from HttpConnection
-                        byte status1 = value[0], status2 = value[1], status3 = value[2];
-                        if (!IsDigit(status1) || !IsDigit(status2) || !IsDigit(status3))
+                        if (name.SequenceEqual(s_statusHeaderName))
+                        {
```

Comment:
Rather than indenting all of this, how about just inverting the condition and doing all the error handling up front, e.g.
```C#
if (_state == StreamState.ExpectingTrailingHeaders)
{
    // Pseudo-headers not allowed in trailers.
    if (NetEventSource.IsEnabled) _connection.Trace("Pseudo-header in trailer headers.");
    throw new HttpRequestException(SR.net_http_invalid_response);
}

if (!name.SequenceEquals(s_statusHeaderName))
{
    if (NetEventSource.IsEnabled) _connection.Trace("Invalid response pseudo-header '{System.Text.Encoding.ASCII.GetString(name)}'.");
    throw new HttpRequestException(SR.net_http_invalid_response);
}

if (_response.StatusCode != 0)
{
    if (NetEventSource.IsEnabled) _connection.Trace("Received duplicate status headers.");
    throw new HttpRequestException(SR.net_http_invalid_response);
}

... // etc.
```